pypodcastparser/Podcast.py

```diff
@@ -353,6 +358,10 @@ def set_published_date(self, tag):
         """Parses published date and set value"""
         try:
             self.published_date = tag.string
+            #Preserve the orignal tag for the start_date
+            self.published_date_string = tag.string
+            pubDate = datetime.datetime.strptime(self.published_date, "%a, %d %b %Y %H:%M:%S %Z")
+            self.published_date = datetime.datetime.strftime(pubDate,"%Y-%d-%m, %H:%M:%S")
```

Comment:
This method is throwing a bunch of errors in our unit tests, think we need to take another look at what we're trying to accomplish here and see if there is a smother solution or if we need to update our Unit Tests.

Reply:
This logic was done to cover the following corner case: 
Here is an exampe of an RSS PubDate: Thu, 05 May 2022 22:11:03 +0000
We have logic in XSL to reformat PubDates to reflect our internal date/time formatting in the start_date and original_air_date fields in Postgres:
2022-05-25 22:11:03


The string pubDate is preserved for logic in the method on line 114 in Item and 154 in Podcasts

The unit tests are passing in my local env: 


(pypod) Patrick-Daleys-MacBook-Pro:pyPodcastParser patrickdaley$ pyenv exec tox
py37 installed: attrs==21.4.0,beautifulsoup4==4.9.1,certifi==2020.6.20,chardet==3.0.4,coverage==6.4.1,datadiff==2.0.0,docopt==0.6.2,httmock==1.4.0,idna==2.10,importlib-metadata==1.7.0,iniconfig==1.1.1,lxml==4.5.2,packaging==21.3,pluggy==0.13.1,py==1.11.0,pyparsing==3.0.9,pytest==7.1.2,pytest-asyncio==0.18.3,requests==2.24.0,soupsieve==2.0.1,tomli==2.0.1,typing_extensions==4.2.0,urllib3==1.25.9,zipp==3.1.0
py37 run-test-pre: PYTHONHASHSEED='2168727786'
py37 run-test: commands[0] | python -m coverage run -m pytest --strict tests
==================================================================================================================== test session starts =====================================================================================================================
platform darwin -- Python 3.7.13, pytest-7.1.2, pluggy-0.13.1
cachedir: .tox/py37/.pytest_cache
rootdir: /Users/patrickdaley/pyPodcastParser
plugins: asyncio-0.18.3
asyncio: mode=legacy
collected 111 items                                                                                                                                                                                                                                          

tests/test_pyPodcastParser.py ..............................................................................................................                                                                                                           [ 99%]
tests/test_test.py .                                                                                                                                                                                                                                   [100%]

env: python 3.7.13

Reply:
@IanNoyes  
I am also curious about the implementation for the corner case below. I added a is_interactive tag in the tag field. Is that correct, or should this be at another layer?

Mapping ihr:interactive field
<is_interactive> is not not a standard RSS field. At this point, only one host service is providing this field specifically for us.
We’ve added mapping in XSL to map the ihr:interactive to our <is_interactive> field.

Reply:
The place you put is_interactive is correct.  Good work!

Instead of trying to parse the dates by hand maybe look into using a library like [dateutil](https://github.com/dateutil/dateutil/tree/6b035517571e63b6a63a493740c5506ec1e5da44) to handle it.